core/classifiers/classifier_utils_test.py

```diff
@@ -137,3 +137,68 @@ def check_that_unicode_validator_works_as_expected(self):
         with self.assertRaisesRegexp(
             Exception, 'Expected \'f\' to be unicode but found str.'):
             classifier_utils.unicode_validator_for_classifier_data(test_dict)
+
+    def test_convert_float_numbers_to_string_in_classifier_data(self):
+        """Make sure that all values are converted correctly."""
+        test_dict = {
+            'a': {
+                'ab': 'abcd',
+                'ad': {
+                    'ada': 'abcdef',
+                    'adc': ['fghi', {
+                        'adca': 'abcd',
+                        'adcb': 0.1234,
+                        'adcc': ['ade', 'afd']
+                    }]
+                },
+                'ae': [['123', '0.123'], ['abc']],
+                'af': [['abcde'], [0.871]]
```

Comment:
You won't ever get a case like this, right?

Reply:
Hard to say! As of now it doesn't seem possible to have a case like this (and as new classifiers are implemented and we keep them under check then maybe we won't ever need a case like this). But with ML classifiers nothing is impossible. The trained model of classifiers is as varied as the ML architectures are.

Reply:
Ah ok! Maybe we can force a definition like the one stated here?

https://github.com/oppia/oppia-ml/pull/26#discussion_r202879447

Also, perhaps, as part of the procedure of incorporating a new classifier, we should tell the developer: put a sample of your classifier data dict in this list (or in this subfolder). Then we would have a backend test that loops over each classifier type and runs the structure check on the given sample.